src/test/java/seedu/address/logic/commands/ClusterCommandTest.java

```diff
@@ -0,0 +1,116 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+import static seedu.address.testutil.TypicalPersons.BENSON;
+import static seedu.address.testutil.TypicalPersons.CARL;
+import static seedu.address.testutil.TypicalPersons.DANIEL;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.AddressDiagnosisStatusPredicate;
+
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code FindCommand}.
+ */
+public class ClusterCommandTest {
+
+    private final Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private final Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        int firstClusterSize = 4;
+        AddressDiagnosisStatusPredicate firstPredicate = new AddressDiagnosisStatusPredicate(
+                "Woodlands", "coronavirus", "UNWELL");
+        ClusterCommand clusterFirstCommand = new ClusterCommand(firstClusterSize, firstPredicate);
+
+        int secondClusterSize = 8;
+        AddressDiagnosisStatusPredicate secondPredicate = new AddressDiagnosisStatusPredicate(
+                "Pasir Ris", "dengue", "PENDING");
+        ClusterCommand clusterSecondCommand = new ClusterCommand(secondClusterSize, secondPredicate);
+
+        // same object -> returns true
+        assertEquals(clusterFirstCommand, clusterFirstCommand);
+
+        // same values -> returns true
+        ClusterCommand clusterFirstCommandCopy = new ClusterCommand(firstClusterSize, firstPredicate);
+        assertEquals(clusterFirstCommand, clusterFirstCommandCopy);
+
+        // different types -> returns false
+        assertNotEquals(1, clusterFirstCommand);
+
+        // null -> returns false
+        assertNotEquals(null, clusterFirstCommand);
+
+        // different cluster size, address, illness and status -> returns false
+        assertNotEquals(clusterFirstCommand, clusterSecondCommand);
+    }
```

Comment:
A little nitpicky but might be better to split this into two methods, one for expected success, and one for expected failure for code coverage and readability reasons